openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -519,18 +538,46 @@ def _create_account(self, request, data):
                 field: [{"user_message": error} for error in error_list]
                 for field, error_list in err.message_dict.items()
             }
-            response = self._create_response(errors, status_code=400)
+            response = self._create_response(request, errors, status_code=400)
         except PermissionDenied:
             response = HttpResponseForbidden(_("Account creation not allowed."))
 
         return response, user
 
-    def _create_response(self, response_dict, status_code):
+    def _create_response(self, request, response_dict, status_code):
         if status_code == 200:
             # keeping this `success` field in for now, as we have outstanding clients expecting this
             response_dict['success'] = True
+        else:
+            self._log_validation_errors(request, response_dict, status_code)
+
         return JsonResponse(response_dict, status=status_code)
 
+    def _log_validation_errors(self, request, errors, status_code):
+        if not REGISTRATION_FAILURE_LOGGING_FLAG.is_enabled():
+            return
+
+        try:
+            user_agent = request.META.get('HTTP_USER_AGENT', '').lower()
+            if 'iphone' in user_agent:
+                agent_category = 'ios'
+            elif 'android' in user_agent:
+                agent_category = 'android'
+            else:
+                agent_category = 'other'
+            for field_key, errors in errors.items():
+                for error in errors:
+                    log.info(
+                        'status_code=%d, agent_category=%s, field=%s, error=%s',
+                        status_code,
+                        agent_category,
+                        field_key,
+                        error['user_message']
+                    )
+        except:  # pylint: disable=bare-except
+            log.exception("Failed to log registration validation error")
+            pass
```

Comment:
Can you add a single unit test in https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_authn/views/tests/test_register.py  to cover this exception path?  You could possibly use/bootstrap an existing test that tests for a `400` status code.

Reply:
I don't actually expect this exception path to ever get hit... I just wanted to be extra confident that the request doesn't fail due to an unexpected edge case.

Reply:
Oops. By exception path, I meant a test to cover L560-569.  Essentially, adding an error-checking test with 
`REGISTRATION_FAILURE_LOGGING_FLAG` enabled.

_(It didn't help that I added my comment on L572.)_

Reply:
Ah got it. Done!